Revamped Overview Articles (#1131)
telerik/blazor-docs: components/editor/overview.md

> The <a href = "https://www.telerik.com/blazor-ui/editor" target="_blank">Blazor HTML Editor component</a> enables your users to create rich textual content through a What-You-See-Is-What-You-Get (WYSIWYG) interface and delivers a set of tools for creating, editing, and formatting text, paragraphs, lists, and other HTML elements.

Reviewer: The Validation section should be split into two parts. The part about SignalR message size should be similar to https://docs.telerik.com/blazor-ui/components/fileselect/overview#large-file-support and have its own heading.

![Editor first look](images/editor-overview.png)

## Get/Set Content

The Blazor HTML Editor interacts with its content (value) like all standard components - through its `Value` parameter. You can use it to get and set the HTML string the editor will work with. [Read more about value binding and data binding...]({%slug get-started-value-vs-data-binding%})

>important @[template](/_contentTemplates/editor/general.md#app-must-sanitize-content)

## Validation

You can use the standard Data Annotation attributes to validate the content of the Editor. For the performance reasons listed above, validation happens with the `DebounceDelay` delay, not immediately on every keystroke, like simpler inputs. [See the Validation article for an example on how to validate the content of the Editor...]({%slug common-features/input-validation%}#editor)

> ## Large File Support 

Reviewer: ```suggestion
## Large Content Support 
```

![Editor first look](images/editor-overview.png)

> ## Dependencies

Reviewer: Apart from linking ProseMirror, this section has unclear benefit to the customer. Consider shortening it. We use a lot of JavaScript in lots of places already.

The Blazor HTML Editor interacts with its content (value) like all standard components - through its `Value` parameter. You can use it to get and set the HTML string the editor will work with. You can read more about value binding and data binding [here]({%slug get-started-value-vs-data-binding%}).

#### You can use the following features to get or set the editor content:

@[template](/_contentTemplates/common/parameters-table-styles.md#table-layout)

> | Parameter | Type and Default value | Description |

Reviewer: Missing parameters, wrong section heading.

|-----------|------------------------|-------------|

> | `@bind-Value`  | `EventCallback<string>` | The recommended approach of using two-way binding to get and set the content of the editor. It lets your view-model provide the initial value, and it will update the view-model as the user alters the HTML. |

Reviewer: `@bind-Value` is not a parameter. `Value` is a parameter and `@bind` is a directive. We can just mention which parameters support two-way binding and link to the respective `...Changed` event.

| `DebounceDelay`  | `int` <br /> `100ms` | The time in milliseconds that passes between updates on the `Value`. The default is `100ms` and if that causes performance issues with many repaints on your view, you can increase it. Since the editor is expected to handle longer editing sessions and larger content than regular inputs, we added this parameter to debounce the view-model updates and events. |

> | `ValueChanged`  | `EventCallback` | Allows you to receive the Editor value and act upon it. If you use the `ValueChanged` event (one-way binding), you can effectively cancel the user's input by not updating the view-model, or you can even alter it with something else. |

Reviewer: We normally don't discuss events in the Overview page.

> ## Validation

Reviewer: The Validation section should be split into two parts. The part about SignalR message size should be similar to https://docs.telerik.com/blazor-ui/components/fileselect/overview#large-file-support and have its own heading.

Images in the content area of the Editor are resizable. To grab the resize handles, hover on the borders of the image.

## Dependencies

The Telerik UI for Blazor Editor uses the ProseMirror engine and it depends on it. You do not need to add any extra assets or references yourself, though, we have taken care of everything internally.

## Editor Parameters

The following table lists Editor parameters, which are not discussed elsewhere in the component documentation. 

@[template](/_contentTemplates/common/parameters-table-styles.md#table-layout)

| Parameter | Type and Default value | Description |
|-----------|------------------------|-------------|

> | `Value`  | `string` | Specifies the value of the component. You can use it for `two-way binding` with the [`ValueChanged` event]({%slug editor-events%}#valuechanged). |

Reviewer: ```suggestion
| `Value`  | `string` | The value of the component. It supports two-way binding or alternatively, use it with the [`ValueChanged` event]({%slug editor-events%}#valuechanged). |
```

| `DebounceDelay`  | `int` <br /> (`100`) | The time in milliseconds that passes between updates on the `Value`. The default is `100ms` and if that causes performance issues with many repaints on your view, you can increase it. Since the editor is expected to handle longer editing sessions and larger content than regular inputs, we added this parameter to debounce the view-model updates and events. |
| `Adaptive`  | `bool` | Defines if the [toolbar]({%slug editor-toolbars%}) should adapt to changes in the width of the component and automatically hide and show the overflowing items in a popup. |
| `Width`  | `string` | Defines the width of the Editor. The default width is `null` but the themes apply `100%`. |
| `Height`  | `string` <br /> (`250px`) | Defines the height of the Editor. |
| `AriaLabelledBy`  | `string` | Maps to the `area-labelledby` attribute. Use this parameter to reference another element to define its accessible name. |
| `AriaDescribedBy`  | `string` | Maps to the `area-describedby` attribute. Use this parameter to establish a relationship between widgets or groups and the text that describes them. |

> ## Component Reference and Methods

Reviewer: ```suggestion
## Editor Reference and Methods
```
